apps/aefate/src/aefa_fate.erl

```diff
@@ -614,15 +622,17 @@ pop_call_stack(ES) ->
             ES2 = check_return_type_protected(Protected, RetType, TVars, Stores, API, ES1),
             pop_call_stack(ES2);
         {local, Function, TVars, BB, ES1} ->
-            ES2 = set_local_function(Function, ES1),
+            % Allow returning to init
+            ES2 = set_local_function(Function, true, ES1),
```

Comment:
Is it possible to manipulate the callstack to change the return function?
Foo -> Bar -> Hack -> NotBar
I don't think it's the case but I want to be fully sure.

Reply:
So if this would be the case it would ofcourse affect not only init. Anyway.

As I understand our callstack assures strict separation of functions even inside one contract. `call_stack` is not accessible in any direct way. It's only changed by the VM when a call is made or we are returning. We have also `accumulator` and `accumulator_stack` that are saved on each call to make sure callee can't affect calculations of the caller.